src/renderer/backend-com.ts

```diff
@@ -0,0 +1,74 @@
+import {
+  BaseDeltaChat,
+  yerpc,
+} from 'deltachat-node/deltachat-jsonrpc/typescript/src/lib'
+import { Message } from 'deltachat-node/deltachat-jsonrpc/typescript/generated/jsonrpc'
+import { DeltaBackend } from './delta-remote'
+import { runtime } from './runtime'
+import { getLogger } from '../shared/logger'
+
+export * as Type from 'deltachat-node/deltachat-jsonrpc/typescript/generated/types'
+
+const { BaseTransport } = yerpc
+const log = getLogger('renderer/jsonrpc')
+
+class ElectronTransport extends BaseTransport {
+  constructor() {
+    super()
+    window.electron_functions.ipcRenderer.on(
+      'json-rpc-message',
+      (_ev, response) => {
+        const message: Message = JSON.parse(response)
+        //   log.debug("received: ", message)
+        this._onmessage(message)
+      }
+    )
+  }
+  _send(message: Message): void {
+    const serialized = JSON.stringify(message)
+    window.electron_functions.ipcRenderer.invoke('json-rpc-request', serialized)
+    log.debug('sent: ', message)
+  }
+}
+
+class ElectronDeltachat extends BaseDeltaChat<ElectronTransport> {
+  close() {
+    /** noop */
+  }
+  constructor() {
+    super(new ElectronTransport())
+  }
+}
+
+export const BackendRemote: BaseDeltaChat<any> = new ElectronDeltachat()
+
+/** Functions with side-effects */
+export namespace EffectfulBackendActions {
```

Comment:
Why the seperation? Also the other backend actions also do have side effects of course.

Reply:
eventually those functions should be transferred to the screen context / screen controller.
They call electron / runtime stuff so I figured to put them into their own functions.
of course we can discuss naming, though that would become unnecessary if we move the functions anyway?